### PR TITLE
add Column.is_in

### LINKED ISSUE #159: Column.is_in
In the last call it was brought up that perhaps we should add `Column.is_in`

This can be achieved with the existing Standard methods:
```python
result = column == values[0]
for value in values[1:]:
    if math.isnan(value):
        result |= column.isnan()
    elif value is namespace.null:
        result |= column.isnull()
    else:
        result |= column == value
```
so it really just be a convenience method. No objections to adding it anyway

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #159

### CODE REVIEW

File: spec/API_specification/dataframe_api/column_object.py [hunk 1]
```diff
@@ -384,3 +384,19 @@ def isnan(self) -> Column:
...
         In particular, does not check for `np.timedelta64('NaT')`.
         """
+
+    def is_in(self, values: Sequence[object]) -> Column[bool]:
+        """
+        Indicate whether the value at each row matches any value in `values`.
+
+        Parameters
+        ----------
+        values : Sequence[object]
```
Comment (jorisvandenbossche) [inline spec/API_specification/dataframe_api/column_object.py:394]: I suppose we can be stricter here, and require that this is a Column object?

File: spec/API_specification/dataframe_api/column_object.py [hunk 2]
```diff
@@ -384,3 +384,19 @@ def isnan(self) -> Column:
...
+            Values to compare against. May include ``float('nan')`` and
+            ``namespace.null``.
```
Comment (jorisvandenbossche) [inline spec/API_specification/dataframe_api/column_object.py:396]: We might need to specify the behaviour for NaN and null (that those give a "match", i.e. resulting in a True value, in contrast to standard equality with `==`)

File: spec/API_specification/dataframe_api/column_object.py (hunk 1 above)
Comment (MarcoGorelli) [reply spec/API_specification/dataframe_api/column_object.py]: yeah sounds fine, thanks!

**Review by kkraus14 — APPROVED** [review]

File: spec/API_specification/dataframe_api/column_object.py [hunk 3]
```diff
@@ -384,3 +384,21 @@ def isnan(self) -> Column:
...
         In particular, does not check for `np.timedelta64('NaT')`.
         """
+
+    def is_in(self, values: Column) -> Column[bool]:
+        """
+        Indicate whether the value at each row matches any value in `values`.
+
+        Parameters
+        ----------
+        values : Column
```
Comment (jbrockmendel) [inline spec/API_specification/dataframe_api/column_object.py:394]: should the dtypes match?

Comment (MarcoGorelli) [reply spec/API_specification/dataframe_api/column_object.py:394]: yeah probably, else pandas does magic stuff like
```python
In [3]: pd.date_range('2000', periods=3).to_series().isin(['2000-01-01'])
Out[3]:
2000-01-01     True
2000-01-02    False
2000-01-03    False
Freq: D, dtype: bool
```
which other libraries almost certainly won't do

Comment (jbrockmendel) [reply spec/API_specification/dataframe_api/column_object.py:394]: btw I think we (pandas) should deprecate that casting

Comment (MarcoGorelli) [reply spec/API_specification/dataframe_api/column_object.py:394]: yeah that'd be good

File: spec/API_specification/dataframe_api/column_object.py [hunk 4]
```diff
@@ -384,3 +384,22 @@ def isnan(self) -> Column:
...
+            Contains values to compare against. May include ``float('nan')`` and
+            ``namespace.null``, which which case ``'nan'`` and ``null`` will
```
Comment (rgommers) [inline spec/API_specification/dataframe_api/column_object.py:396]: can you leave off `namespace`? Just plain `null` should do fine here.

File: spec/API_specification/dataframe_api/column_object.py (hunk 3 above)
Comment (rgommers) [reply spec/API_specification/dataframe_api/column_object.py:440]: I'll resolve this - perhaps you want to turn the above "we should deprecate this in pandas" into a pandas issue @jbrockmendel or @MarcoGorelli?

**Pull request merged**